gradio/blocks.py

```diff
@@ -1673,6 +1675,8 @@ async def process_api(
                 state,
                 limiter=self.limiter,
             )
+            if root_path is not None:
```

Comment:
This also needs to be applied to the batch case, no?

Reply:
Batch mode is a bit tricky, because we have to add the root_urls to each element, which in theory can be different (though unlikely). For now, I'll just assume the root_urls are the same.